### PR TITLE
🏇 Improving Personal Profile: Adding the data model

### LINKED ISSUE #37: 🏇 Data model for the Personal Profile
Tasks:

- [x]  Create the basic data model - personal profile
- [x]  Adding the ability to upload a profile picture
- [x]  Adding the ability to upload a resume as a PDF
- [x]  Adding tests to the data model

The data model in the ERD:

image

### PR DESCRIPTION
**Pull request by ShirleyFichman**
Added the Personal Profile data model.

Did the following:

- Created a Personal Profile data model
- run `makemigrations` and `migrate`
- Registered the new data model to the admin panel
- Added the ability to upload a profile picture & resume
- Added tests to the data model

Depends on #35 
Close #37

### CODE REVIEW

**Comment by ShirleyFichman** [comment]
Please note this PR is instead of #28 which can be closed now, thank you

**Comment by taljacob2** [comment]
> Please note this PR is instead of #28 which can be closed now, thank you

Done.

File: personal_profile/models.py [hunk 1]
```diff
@@ -14,6 +14,6 @@ class PersonalProfile(models.Model):
     about= models.TextField() 
     birth_date= models.DateField()
     profile_pic= models.ImageField(null=True, blank=True)
-  
+
```
Comment (kasemAlem) [inline personal_profile/models.py:18]: What changes were made in here ?

Comment (ShirleyFichman) [reply personal_profile/models.py:18]: @kasemAlem the flask8 asked to have a newline at the end of file with no space

**Comment by Yarboa** [comment]
@ShirleyFichman 
Sorry I am lost, 
I would recommend to focus only the model part, either in this PR or new one,
This one is too big to review for now

**Comment by taljacob2** [comment]
I am currently cleaning this PR a bit.

**Comment by taljacob2** [comment]
UPDATE: Resolved conflicts with `main`.

**Comment by taljacob2** [comment]
Made a `rebase fixup` to all "*Flake8*-syntax-related" commits on:
[2d59b83e3b108563e592ef0bd98391b303b0bbfc](https://github.com/redhat-beyond/JobSeeker/pull/36/commits/2d59b83e3b108563e592ef0bd98391b303b0bbfc)

**Comment by taljacob2** [comment]
Hi all!
I'm done polishing this PR for now!

## What I've done so far:

- [Made a rebase fixup to all "Flake8-syntax-related" commits on:
   2d59b83e3b108563e592ef0bd98391b303b0bbfc](https://github.com/redhat-beyond/JobSeeker/pull/36#issuecomment-1076021457).
- [Resolved conflicts with `main`](https://github.com/redhat-beyond/JobSeeker/pull/36#issuecomment-1076007997).
- Added a simple `.html` to view the application. In [3affcc1](https://github.com/redhat-beyond/JobSeeker/pull/36/commits/3affcc1de8e0c9627d3cde43d52cb0a164a10192).
  > NOTE: This closes #52.

## Left TODO:
- Add `pytest`.

**Review by Yarboa — COMMENTED** [review]
1) Need to rebase -> push
2) Please ping other team members to review
3) As far as i can tell, you should have tests for the model.
4) Adding data migrations
https://docs.djangoproject.com/en/4.0/topics/migrations/#data-migrations-1

Please refer contribution.md

File: Pipfile [hunk 2]
```diff
@@ -8,6 +8,7 @@ django = "*"
 
 [dev-packages]
 flake8 = "*"
+pillow = "*"
```
Comment (amittcohen) [inline Pipfile:11]: Can you explain this line?

Comment (taljacob2) [reply Pipfile:11]: This is a plugin @ShirleyFichman is using in order to save images to the database.

**Comment by Yarboa** [comment]
Please rebase and push

**Comment by taljacob2** [comment]
> Please rebase and push

In progress.

**Comment by taljacob2** [comment]
> > Please rebase and push
> 
> In progress.

Done.
Rebased to `main` + fixed the migrations of `personal_profile` in [585271a](https://github.com/redhat-beyond/JobSeeker/commit/585271a92c4d7d89feeec24df9127dcbf8ca1f9f).

### Left TODO:
- Add tests.

**Comment by Yarboa** [comment]
> 1. Need to rebase -> push
> 2. Please ping other team members to review
> 3. As far as i can tell, you should have tests for the model.

@ShirleyFichman 
I do not see any tests, 
PR could not b approved, sorry

> 4. Adding data migrations
>    https://docs.djangoproject.com/en/4.0/topics/migrations/#data-migrations-1
> 
> Please refer contribution.md

**Comment by ShirleyFichman** [comment]
> > 1. Need to rebase -> push
> > 2. Please ping other team members to review
> > 3. As far as i can tell, you should have tests for the model.
> 
> @ShirleyFichman I do not see any tests, PR could not b approved, sorry
> 
> > 4. Adding data migrations
> >    https://docs.djangoproject.com/en/4.0/topics/migrations/#data-migrations-1
> > 
> > Please refer contribution.md

@Yarboa 
I added an "empty" data migrations since I didn't understand what is the logic that I needed to implement on the function for RunPython.
May you please elaborate? Thank you!

**Comment by Yarboa** [comment]
> > > 1. Need to rebase -> push
> > > 2. Please ping other team members to review
> > > 3. As far as i can tell, you should have tests for the model.
> > 
> > 
> > @ShirleyFichman I do not see any tests, PR could not b approved, sorry
> > > 4. Adding data migrations
> > >    https://docs.djangoproject.com/en/4.0/topics/migrations/#data-migrations-1
> > > 
> > > Please refer contribution.md
> 
> @Yarboa I added an "empty" data migrations since I didn't understand what is the logic that I needed to implement on the function for RunPython. May you please elaborate? Thank you!

Please take a look here
Maybe it will help you to understand
https://github.com/Yarboa/beyond-tutorial/blob/main/msgboard/migrations/0002_test_data.py

It is populating data in DB, this way during the Model creation you verify
Model class constructed properly

**Comment by Yarboa** [comment]
Please rebase push

**Comment by ShirleyFichman** [comment]
> Please rebase push

Done

**Comment by ShirleyFichman** [comment]
I updated via Slack on the latest problems I'm facing with PyTest:
Many different app tests fail although I didn't change them.
Also, one failure of the Personal profile is because it expects to receive STR instead of DATE which is an incorrect expectation, the test should pass.

File: personal_profile/resources/personal_profiles.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+from django.db import models
+PERSONAL_PROFILES = [
+    [
+        'Microsoft',
+        'Bil Gates',
+        'Im a co-founder of microsoft, and have been the richest guy in the world for a while...',
+        models.DateField(1955, 10, 28),
```
Comment (paOmer) [inline personal_profile/resources/personal_profiles.py:7]: In all of those DateField change it to the following: `datetime.date(1955, 10, 28)`, with a date to your choice.
Also replace the current import on the top of the page `from django.db import models` to `import datetime`

**Review by paOmer — CHANGES_REQUESTED** [review]

File: personal_profile/models.py [hunk 4]
```diff
@@ -1,3 +1,21 @@
...
+class PersonalProfile(models.Model):
+    company = models.CharField(max_length=100)
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    about = models.TextField()
+    birth_date = datetime.date(2000, 12, 10)
+    profile_pic = models.ImageField(null=True, blank=True)
+    resume = models.FileField(null=True, upload_to='resumes')
+
+    def __str__(self):
+        return f'{self.user.username} Profile'
```
Comment (Yarboa) [inline personal_profile/models.py:21]: @ShirleyFichman please take a look in the flake8 errors
No new line here

File: personal_profile/resources/personal_profiles.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        'Im a co-founder of microsoft, and have been the richest guy in the world for a while...',
+        datetime.date(1955, 10, 28),
+        'personal_profile/static/personal_profile/images/profile_pics/billgates.jpg',
+        'personal_profile/static/personal_profile/images/resumes/billgates_resume.jpg',
+    ],
+    [
+        'Apple',
+        'Tim Cook',
+        'Im the CEO of Apple, and honestly Im just the guy everyone calls Steve Jobs',
+         datetime.date(1960, 11, 1),
```
Comment (Yarboa) [inline personal_profile/resources/personal_profiles.py:15]: Indentation Error, please delete one space

Comment (ShirleyFichman) [reply personal_profile/resources/personal_profiles.py]: Done

File: personal_profile/resources/personal_profiles.py (hunk 3 above)
Comment (ShirleyFichman) [reply personal_profile/resources/personal_profiles.py]: Solved.

**Review by taljacob2 — APPROVED** [review]
LGTM!

Can you please add to this PR's description that it closes #52?

**Review by paOmer — APPROVED** [review]
LGTM!

**Review by Yarboa — APPROVED** [review]
LGTM

**Pull request merged**